templates/login.liquid

```diff
@@ -0,0 +1,174 @@
+      }
+      .ea-layout {
+        display: flex;
+        width: auto;
+        flex-direction: column;
+        justify-content: center;
+        align-items: center;
+      }
+      @media (min-width: 480px) {
+        .ea-layout {
+          width: 100%;
+        }
+      }
+      @media (min-width: 1070px) {
+        .ea-layout {
+          padding: 80px 0;
+        }
+        .ea-layout-forum {
+          flex-direction: row;
+          align-items: flex-start;
+        }
+        .ea-layout-contact {
+          flex-direction: column;
+          align-items: center;
+        }
+      }
+      .forum-instructions {
+        width: 100%;
+        border-radius: 5px;
+        background-color: #dae0e2;
+        padding: 20px;
+      }
+      @media (min-width: 480px) {
+        .forum-instructions {
+          width: 400px;
+        }
+      }
+      @media (min-width: 1070px) {
+        .forum-instructions {
+          margin-right: var(--forum-instructions-margin);
+          width: var(--forum-instructions-width);
+        }
+      }
+      .forum-instructions p {
+        margin-bottom: 1em;
+        line-height: 1.5;
+      }
+      .forum-instructions-details-text p:last-child {
+        margin-bottom: 0;
+      }
+      .forum-instructions-hide-this {
+        margin-top: 1em;
+        margin-bottom: 0;
+      }
+      @media (min-width: 1070px) {
+        .forum-instructions-hide-this {
+          display: none;
+        }
+      }
+      .forum-instructions-show-more {
+        margin-top: 1em;
+        margin-bottom: 0;
+        display: none;
+      }
+      .widget-wrapper {
+        width: 100%;
+      }
+      @media (min-width: 480px) {
+        .widget-wrapper {
+          width: 400px;
+        }
+      }
+      @media (max-width: 480px) {
+        ._prompt-box-outer {
+          width: 100vw;
+        }
+      }
+      .forum-instructions-balancer {
+        visibility: hidden;
+      }
+      @media (min-width: 1070px) {
+        .forum-instructions-balancer {
+          width: calc(var(--forum-instructions-width) + var(--forum-instructions-margin));
+        }
+      }
+      .contact-us {
+        width: 100%;
+        height: calc(2em+5px);
+        background-color: #dae0e2;
+        padding: 10px;
+        border-radius: 5px;
+        text-align: center;
+        position: relative;
+        z-index: 5;
+      }
+      @media (min-width: 480px) {
+        .contact-us {
+          margin-top: 20px;
+          margin-bottom: 20px;
+          width: 400px;
+        }
+      }
+      @media (min-width: 1070px) {
+        .contact-us {
+          margin-bottom: 0;
+        }
+      }
+    </style>
+    <script>
+      function hideForumDetails() {
+        const details = document.getElementById("forum-instructions-details");
+        details.style.display = "none";
+        const showMore = document.getElementById("forum-instructions-show-more");
+        showMore.style.display = "block";
```

Comment:
If I was implementing this, I think the pattern I'd use would be to:
- Set an additional class on the parent DIV (i.e. `.forum-instructions`) of `.open`
- Target descendant CSS based on `.forum-instructions.open`. E.g. `.forum-instructions.open .show-open { display: block; }; .forum-instructions.open .hide-open {display: none}` and vice-versa
- Instead of changing styles directly in JS, just change the `.open` or `.closed` class of the parent DIV using [`element.classList.add()/.remove()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList)

This means that you're coupling a minimum amount of display logic to your application code, and instead keeping it in CSS where it belongs.

Reply:
A nice thing about this too is that you can e.g. more easily use CSS transitions

So, if you wanted the panel to collapse, you could do something like:

```html
<div class='forum-instructions open'>
	<div class='show-open collapsible'>
		Lorem ipsum dolor sit amet
	</div>
</div>
```

```css
.forum-instructions .collapsible {
	transition: max-height 0.3s;
}
.forum-instructions.open .collapsible.show-open {
	max-height: 1000px;
}
.forum-instructions.close .collapsible.show-open {
	max-height: 0;
}
```

(Caveat, the above _should_ work, but there will be some delay while the max height animates down from 1000px. In reality you'd probably want to customise this for a specific element, and then target the `.show-open` `max-height` to be approximately whatever the rendered height of the element actually was on mobile)

Reply:
(All of the above said to be instructive about these kinds of patterns in general, not to necessarily suggest you should change anything)

Reply:
I implemented the pattern, mostly for my own edification, but I couldn't immediately get the max-height transition pattern to work. I'll keep the class-based show-hide, but won't try to get the transition to work.